src/components/AddressBadge.tsx

```diff
@@ -17,42 +17,75 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { ComponentPropsWithoutRef } from 'react'
-import styled, { css } from 'styled-components'
-import tinycolor from 'tinycolor2'
+import styled, { useTheme } from 'styled-components'
 
+import { Address } from '../contexts/addresses'
+import { AddressSettings, isAddress, isAddressSettings } from '../utils/addresses'
 import Badge from './Badge'
 
 type AddressBadgeProps = ComponentPropsWithoutRef<typeof Badge> & {
-  addressName: string
-  opaqueBg?: boolean
+  address: Address | AddressSettings
+  truncate?: boolean
 }
 
-const AddressBadge = ({ addressName, className, ...props }: AddressBadgeProps) => (
-  <Badge className={className} rounded {...props}>
-    {addressName}
-  </Badge>
-)
+const AddressBadge = ({ address, className, ...props }: AddressBadgeProps) => {
+  const theme = useTheme()
+
+  let data
+
+  if (isAddress(address)) {
+    data = {
+      color: address.settings.color,
+      isMain: address.settings.isMain,
+      label: address.getLabelName(false)
+    }
+  } else if (isAddressSettings(address)) {
+    data = address
+  } else {
+    data = {
+      color: theme.font.primary,
+      isMain: false,
+      label: ''
+    }
+  }
+
+  return (
+    <div className={className} {...props}>
```

Comment:
hm, are `{...props}` really needed there, now that we don't use `Badge` anymore?

Reply:
I think I thought it was necessary to pass the props so the styled component could see it, but no it's already getting them / typed I realize now